test/ui/ui-main.swapping.test.ts

```diff
@@ -0,0 +1,149 @@
+import { connectPolkadotWallet } from "../../utils/frontend/utils/Handlers";
+
+require("dotenv").config();
+
+jest.spyOn(console, "log").mockImplementation(jest.fn());
+
+jest.setTimeout(1500000);
+let driver: WebDriver;
+process.env.NODE_ENV = "test";
+let sudo: SudoUser;
+let testUser1: User;
+const testAssetName = "TST4";
+let testAssetId: BN;
+
+describe("UI tests - swapping assets", () => {
+  beforeAll(async () => {
+    try {
+      getApi();
+    } catch (e) {
+      await initApi();
+    }
+
+    const keyring = new Keyring({ type: "sr25519" });
+    const node = new Node(getEnvironmentRequiredVars().chainUri);
+    await node.connect();
+    sudo = new SudoUser(keyring, node);
+    keyring.addPair(sudo.keyRingPair);
+    await setupApi();
+    await setupUsers();
+    driver = await DriverBuilder.getInstance();
+    const { mnemonic } = await setupPolkadotExtension(driver);
+    testUser1 = new User(keyring);
+    testUser1.addFromMnemonic(keyring, mnemonic);
+    testAssetId = await createAssetIfMissing(sudo, testAssetName);
+    await createPoolIfMissing(
+      sudo,
+      "1800000000000000000000000000000000",
+      MGA_ASSET_ID,
+      testAssetId
+    );
+
+    await Sudo.batchAsSudoFinalized(
+      Assets.mintToken(new BN(7), testUser1), // transferAll test
+      Assets.mintToken(testAssetId, testUser1), // transferAll test
+      Assets.mintNative(testUser1)
+    );
+
+    testUser1.addAsset(testAssetId);
+    await testUser1.refreshAmounts(AssetWallet.BEFORE);
+  });
+
+  test("User can swap two assets", async () => {
+    getApi();
+
+    const mga = new Mangata(driver);
+    await mga.go();
+    const sidebar = new Sidebar(driver);
+    const noWalletConnectedInfoDisplayed =
+      await sidebar.isNoWalletConnectedInfoDisplayed();
+    expect(noWalletConnectedInfoDisplayed).toBeTruthy();
+
+    await connectPolkadotWallet(driver, sidebar, mga);
+    const isWalletConnected = sidebar.isWalletConnected("acc_automation");
+    expect(isWalletConnected).toBeTruthy();
+
+    const swapView = new Swap(driver);
+    await swapView.toggleSwap();
+    await swapView.toggleShowAllTokens();
+    await swapView.selectPayAsset(MGA_ASSET_NAME);
+    await swapView.selectGetAsset(testAssetName);
+    await swapView.addPayAssetAmount("0.001");
+    const testAssetValue = await swapView.getBalanceFromAssetGet();
+    await swapView.doSwap();
+    const modal = new NotificationModal(driver);
+    const isModalWaitingForSignVisible = await modal.isModalVisible(
+      ModalType.Confirm
+    );
+    expect(isModalWaitingForSignVisible).toBeTruthy();
+    await Polkadot.signTransaction(driver);
+    //wait four blocks to complete the action.
+    const visible: boolean[] = [];
+    for (let index = 0; index < 4; index++) {
+      visible.push(await modal.isModalVisible(ModalType.Progress));
+      await waitNewBlock();
+    }
+    expect(
+      visible.some((visibleInBlock) => visibleInBlock === true)
+    ).toBeTruthy();
```

Comment:
Optional: 
I guess this will be needed in the future, perhaps a utils function?

Reply:
Right! Will include that in next Pr with pools tests.